decisions/0004-host-naming-conventions.md

```diff
@@ -0,0 +1,57 @@
+---
+status: proposed
+date: 2022-12-11
+deciders: Shreyas Cholia
+consulted: {list everyone whose opinions are sought (typically subject-matter experts); and with whom there is a two-way communication}
+informed: {list everyone who is kept up-to-date on progress; and with whom there is a one-way communication}
+---
+# Standardize convention for host names in NMDC
+
+## Context and Problem Statement
+
+Hostnames in the `*.microbiomedata` should have a standard naming pattern in order to 
+- make it easier for humans to remember endpoints
+- enable automation by using a standard pluggable convention 
+- integrate with underlying kubernetes / rancher platform
+- simplify management of certificates
+
+## Decision Drivers
+
+* Restrictions imposed by Spin - Spin has internal names and we would like parity with these if possible
```

Comment:
what are these restrictions (brief summary is fine -- I actually don't know the restrictions)?

Reply:
See updates